src/brod_sock.erl

```diff
@@ -203,6 +206,38 @@ maybe_upgrade_to_ssl(Sock, _Mod = ssl, SslOpts = [_|_], Timeout) ->
 maybe_upgrade_to_ssl(Sock, _Mod, _SslOpts, _Timeout) ->
   {ok, Sock}.
 
+maybe_sasl_auth(_Sock, _Mod, _ClientId, _Timeout, _SaslOpts = undefined) -> ok;
+maybe_sasl_auth(Sock, Mod, ClientId, Timeout,
```

Comment:
Meaning, clients should not block api which is totally legit from the server point of view.